src/test/java/com/nvidia/spark/rapids/jni/GetJsonObjectTest.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.jni;
+
+import ai.rapids.cudf.ColumnVector;
+import org.junit.jupiter.api.Test;
+
+import ai.rapids.cudf.HostColumnVector;
+import static ai.rapids.cudf.AssertUtils.assertColumnsAreEqual;
+
+public class GetJsonObjectTest {
+  /**
+   * Test: query is $.k
+   */
+  @Test
+  void getJsonObjectTest() {
+    JSONUtils.PathInstructionJni[] query = new JSONUtils.PathInstructionJni[2];
+    query[0] = new JSONUtils.PathInstructionJni(JSONUtils.PathInstructionType.KEY, "", -1);
```

Comment:
Do we support arrays yet? I don't see any tests that include an array index that is not -1.

Reply:
Found a big for array index query, I'll fix it.

Reply:
Done